packages/flutter_tools/tool/tool_coverage.dart

```diff
@@ -0,0 +1,113 @@
+/// See: https://github.com/linux-test-project/lcov.git.
+Future<void> main() async {
+  await runInContext(() async {
+    final CoverageCollector coverageCollector = CoverageCollector(
+      flutterProject: await FlutterProject.current(),
+    );
+    final String flutterRoot = Directory.current.parent.parent.path;
+    final String dartPath = path.join(flutterRoot, 'bin', 'cache', 'dart-sdk', 'bin', 'dart');
+    final Pool pool = Pool(Platform.numberOfProcessors);
+    final List<Future<void>> pending = <Future<void>>[];
+
+    Future<void> runTest(File testFile) async {
+      final PoolResource resource = await pool.request();
+      final File fakeTest = File(path.join(tempDirectory.path, testFile.path))
+        ..createSync(recursive: true)
+        ..writeAsStringSync('''
+import "package:test/test.dart";
+import "${path.absolute(testFile.path)}" as entrypoint;
+
+void main() {
+  group('', entrypoint.main);
+}
+''');
+      final int port = await _findPort();
+      final Uri coverageUri = Uri.parse('http://127.0.0.1:$port');
+      final Completer<void> completer = Completer<void>();
+      final Process testProcess = await Process.start(
+        dartPath,
+        <String>[
+          '--packages=${File('.packages').absolute.path}',
+          '--pause-isolates-on-exit',
+          '--enable-asserts',
+          '--enable-vm-service=${coverageUri.port}',
+          fakeTest.path,
+        ],
+        runInShell: true,
+        environment: <String, String>{
+          'FLUTTER_ROOT': Directory.current.parent.parent.path,
+        }).timeout(const Duration(seconds: 30));
+      testProcess.stdout
+        .transform(utf8.decoder)
+        .transform(const LineSplitter())
+        .listen((String line) {
+          print(line);
+          if (line.contains('All tests passed') || line.contains('Some tests failed')) {
+            completer.complete(null);
+          }
+        });
+      try {
+        await completer.future;
+        await coverageCollector.collectCoverage(testProcess, coverageUri).timeout(const Duration(seconds: 30));
+        testProcess?.kill();
+      } on TimeoutException {
+        print('Failed to collect coverage for ${testFile.path} after 30 seconds');
+      } finally {
+        resource.release();
+      }
+    }
+
+    for (FileSystemEntity fileSystemEntity in Directory('test').listSync(recursive: true)) {
+      if (!fileSystemEntity.path.endsWith('_test.dart')) {
+        continue;
+      }
+      pending.add(runTest(fileSystemEntity));
+    }
+    await Future.wait(pending);
+    final String lcovData = await coverageCollector.finalizeCoverage();
+    final String lcovPath = path.join('coverage', 'lcov.info');
+    final String htmlPath = path.join('coverage', 'report.html');
+    File(lcovPath)
+      ..createSync(recursive: true)
+      ..writeAsStringSync(lcovData);
+    await Process.run('genhtml', <String>[lcovPath, '-o', htmlPath,], runInShell: true);
```

Comment:
Depending on what the plans for this script are, I think it will save work later on to provide command line arguments now to override defaults for things like the flutter root, the path to the tests, the path to .packages, the output html file, the path to genhtml, etc.

Reply:
I would hope that this script dies a quick death. Therefore, we should plan for this to exist for at least a year